packages/lib/sdk/src/utils/batchUp.js

```diff
@@ -10,7 +10,8 @@ export const batchUp = (fn, timeout = 200) => {
 	/** @type {Input[]} */
 	const collected = [];
 
-	const finalize = debounce(() => {
+	// todo: possibly algorithmic nightmare at high n on server?
```

Comment:
What about this is algorithmically problematic with a high number of collected items?

Reply:
iirc there was a codepath that ran O(n) work n times for n = number of options that was worked around via `batchUp`, not sure if that's still around though

Reply:
Sounds like the function being used in `batchUp` could potentially be problematic, but not `batchUp` itself?

I think we should remove this log or relocate to the specific codepath you were thinking of

Reply:
[f7ffb48](https://github.com/evidence-dev/evidence/pull/2553/commits/f7ffb4860fcfa4d1d6e8dcc9dc30b942ec0b52dd)